docs/fhir-guide/FHIR 101 - Practical Guide.ipynb

```diff
@@ -935,16 +1061,346 @@
+    "                \"path\": \"url\"\n",
+    "            }\n",
+    "        ],\n",
+    "        \"ordered\": False,\n",
+    "        \"rules\": \"open\"\n",
+    "    }\n",
+    "},\n",
+    "{\n",
+    "    \"path\": \"Patient.extension\",\n",
+    "    \"sliceName\": \"patient-race\",\n",
+    "    \"short\": \"Race\",\n",
+    "    \"definition\": \"Race of patient\",\n",
+    "    \"min\": 0,\n",
+    "    \"max\": \"1\",\n",
+    "    \"type\": [\n",
+    "        {\n",
+    "            \"code\": \"Extension\",\n",
+    "            \"profile\": [\n",
+    "                \"http://fhir.kids-first.io/StructureDefinition/patient-race\"\n",
+    "            ]\n",
+    "        }\n",
+    "    ],\n",
+    "    \"isModifier\": False\n",
+    "}\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "#### Explanation of Parts\n",
+    "Let's break down each piece. First we'll start with the piece that says we want to use extensions in the `Patient` `StructureDefinition`:\n",
+    "```json\n",
+    "{\n",
+    "    \"path\": \"Patient.extension\",  \n",
+    "    \"slicing\": {\n",
+    "        \"discriminator\": [\n",
+    "            {\n",
+    "                \"type\": \"value\",\n",
+    "                \"path\": \"url\"\n",
+    "            }\n",
+    "        ],\n",
+    "        \"ordered\": False,\n",
+    "        \"rules\": \"open\"\n",
+    "    }\n",
+    "}\n",
+    "```\n",
+    "- `path`: All extensions go inside an `extension` element in the Patient resource\n",
+    "- `slicing`: The `Patient.extension` is a list of extensions and each one must be able to be uniquely identified in that list. The extension's URL is used to \"slice\" or identify the extension in the list.\n",
+    "- `ordered=False`: order of extensions does not matter in a Patient resource\n",
+    "- `rules=open`: additional extensions can be added in a Patient resource\n",
+    "\n",
+    "The next part is the actual binding of the race extension to the `Patient` `StructureDefinition`: \n",
+    "\n",
+    "```json\n",
+    "{\n",
+    "    \"path\": \"Patient.extension\",\n",
+    "    \"sliceName\": \"race\",\n",
+    "    \"short\": \"Race\",\n",
+    "    \"definition\": \"Race of patient\",\n",
+    "    \"min\": 0,\n",
+    "    \"max\": \"1\",\n",
+    "    \"type\": [\n",
+    "        {\n",
+    "            \"code\": \"Extension\",\n",
+    "            \"profile\": [\n",
+    "                \"http://fhir.kids-first.io/StructureDefinition/patient-race\"\n",
+    "            ]\n",
+    "        }\n",
+    "    ],\n",
+    "    \"isModifier\": False\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "The important parts to note here are: \n",
+    "\n",
+    "- `min/max`: this says Patient instances can only have one occurrence of the `race` extension and it is not required\n",
+    "- `type`: this binds the Patient's race attribute to it's `StructureDefinition`\n"
+   ]
```

Comment:
The explanation about `"isModifier": False` is missing here.

Reply:
lol i have no idea what it means ... i'll have to figure it out